websearch/websearch/src/durability.rs

```
@@ -103,46 +120,60 @@ mod durable_impl {
                     }
                 });
 
-                let persisted_params = result?;
-                durability.persist_infallible(
-                    StartSearchInput {
-                        params: persisted_params.clone(),
-                    },
-                    persisted_params.clone(),
-                );
-                Ok(SearchSession::new(DurableSearchSession::<Impl>::live(
-                    Impl::unwrapped_search_session(persisted_params).unwrap(),
-                )))
+                match result {
+                    Ok(persisted_params) => {
+                        durability
+                            .persist(persisted_params.clone(), Ok(persisted_params.clone()))?;
+                        Ok(SearchSession::new(DurableSearchSession::<Impl>::live(
+                            Impl::unwrapped_search_session(persisted_params).unwrap(),
```

avoid using unwrap. we should not be panicing, but rather return the error.